homeassistant/components/mqtt/light/schema_template.py

```diff
@@ -184,9 +179,9 @@ def _update_color_mode(self) -> None:
     def _state_received(self, msg: ReceiveMessage) -> None:
         """Handle new MQTT messages."""
         state = self._value_templates[CONF_STATE_TEMPLATE](msg.payload)
-        if state == STATE_ON:
+        if state == LightState.ON:
```

Comment:
This shouldn't use the state enum as it's not comparing an entity state. It's comparing a template rendered MQTT message payload.